ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -106,27 +106,29 @@ private[apiserver] final class ApiSubmissionService private[services] (
       request: SubmitRequest
   )(implicit telemetryContext: TelemetryContext): Future[Unit] =
     withEnrichedLoggingContext(logging.commands(request.commands)) { implicit loggingContext =>
-      logger.info("Submitting transaction")
-      logger.trace(s"Commands: ${request.commands.commands.commands}")
-
-      implicit val contextualizedErrorLogger: ContextualizedErrorLogger =
+      // Spin up a Future so that all the work is done not on the dispatcher thread
+      // but using the pool provided with the ExecutionContext of this class instance.
+      Future {
```

Comment:
This is great workaround. Please add a TODO or NOTE in the code for two reasons:
1. We still don't know WHY do we need to do this. I think answering this question might be insightful at some point. 
2. To make sure no one touching this code feels that it needs the optimization which brings us to point zero :trollface: